rectools/models/sasrec.py

```diff
@@ -52,6 +55,100 @@ def forward(self, seqs: torch.Tensor, timeline_mask: torch.Tensor, attn_mask: to
         raise NotImplementedError()
 
 
+class SessionEncoderDataPreparatorBase:
+    """Base class for data preparator. Used only for type hinting."""
+
+    def __init__(
+        self,
+        session_maxlen: int,
+        batch_size: int,
+        item_extra_tokens: tp.Sequence[tp.Hashable] = (PADDING_VALUE,),
+        shuffle_train: bool = True,  # not shuffling train dataloader hurts performance
+        train_min_user_interactions: int = 2,
+    ) -> None:
+        self.session_maxlen = session_maxlen
+        self.batch_size = batch_size
+        self.item_extra_tokens = item_extra_tokens
+        self.shuffle_train = shuffle_train
+        self.train_min_user_interactions = train_min_user_interactions
+        self.item_id_map: IdMap
+        # TODO: add SequenceDatasetType for fit and recommend
+
+    def get_known_items_sorted_internal_ids(self) -> np.ndarray:
+        """TODO"""
+        return self.item_id_map.get_sorted_internal()[self.n_item_extra_tokens :]
+
+    def get_known_item_ids(self) -> np.ndarray:
+        """TODO"""
+        return self.item_id_map.get_external_sorted_by_internal()[self.n_item_extra_tokens :]
+
+    @property
+    def n_item_extra_tokens(self) -> int:
+        """TODO"""
+        return len(self.item_extra_tokens)
+
+    def process_dataset_train(self, dataset: Dataset) -> Dataset:
+        """TODO"""
+        raise NotImplementedError()
+
+    def get_dataloader_train(self, processed_dataset: Dataset) -> DataLoader:
+        """TODO"""
+        raise NotImplementedError()
+
+    def get_dataloader_recommend(self, dataset: Dataset) -> DataLoader:
+        """TODO"""
+        raise NotImplementedError()
+
+    def transform_dataset_u2i(self, dataset: Dataset, users: ExternalIds) -> Dataset:
+        """TODO"""
+        raise NotImplementedError()
+
+    def transform_dataset_i2i(self, dataset: Dataset) -> Dataset:
+        """TODO"""
+        raise NotImplementedError()
+
+
+class SessionEncoderLightningModuleBase(LightningModule):
```

Comment:
Please move this class under `# ####  --------------  Lightning Model  --------------  #### #`

Reply:
fixed